waspc/data/Generator/templates/react-app/src/auth/forms/Auth.tsx

```diff
@@ -42,590 +29,67 @@ const HeaderText = styled('h2', {
-  borderRadius: '0.375rem',
-  marginTop: '1rem',
-  background: '$gray400',
-})
 
-const ErrorMessage = styled(Message, {
-  background: '$errorBackground',
-  color: '$errorText',
+export const AuthContext = createContext({
+  isLoading: false,
+  setIsLoading: (isLoading: boolean) => {},
+  setErrorMessage: (errorMessage: ErrorMessage | null) => {},
+  setSuccessMessage: (successMessage: string | null) => {},
 })
 
-const SuccessMessage = styled(Message, {
-  background: '$successBackground',
-  color: '$successText',
-})
-
-{=# isGoogleAuthEnabled =}
-const googleSignInUrl = `${config.apiUrl}{= googleSignInPath =}`
-{=/ isGoogleAuthEnabled =}
-{=# isGitHubAuthEnabled =}
-const gitHubSignInUrl = `${config.apiUrl}{= gitHubSignInPath =}`
-{=/ isGitHubAuthEnabled =}
-
 function Auth ({ state, appearance, logo, socialLayout = 'horizontal' }: {
     state: State;
 } & CustomizationOptions) {
-  const isLogin = state === "login";
   const [errorMessage, setErrorMessage] = useState<ErrorMessage | null>(null);
   const [successMessage, setSuccessMessage] = useState<string | null>(null);
   const [isLoading, setIsLoading] = useState(false);
-  {=# isAnyPasswordBasedAuthEnabled =}
-  const history = useHistory();
-  const onErrorHandler = (error) => {
-    setErrorMessage({ title: error.message, description: error.data?.data?.message })
-  };
-  {=/ isAnyPasswordBasedAuthEnabled =}
-  {=# isUsernameAndPasswordAuthEnabled =}
-  const { handleSubmit, usernameFieldVal, passwordFieldVal, setUsernameFieldVal, setPasswordFieldVal } = useUsernameAndPassword({
-    isLogin,
-    onError: onErrorHandler,
-    onSuccess() {
-      // Redirect to configured page, defaults to /.
-      history.push('{= onAuthSucceededRedirectTo =}')
-    },
-  });
-  {=/ isUsernameAndPasswordAuthEnabled =}
-  {=# isEmailAuthEnabled =}
-  const { handleSubmit, emailFieldVal, passwordFieldVal, setEmailFieldVal, setPasswordFieldVal } = useEmail({
-    isLogin,
-    onError: onErrorHandler,
-    showEmailVerificationPending() {
-      setSuccessMessage(`You've signed up successfully! Check your email for the confirmation link.`)
-    },
-    onLoginSuccess() {
-      // Redirect to configured page, defaults to /.
-      history.push('{= onAuthSucceededRedirectTo =}')
-    }
-  });
-  {=/ isEmailAuthEnabled =}
-  {=# isAnyPasswordBasedAuthEnabled =}
-  async function onSubmit (event: FormEvent<HTMLFormElement>) {
-    event.preventDefault();
-    setIsLoading(true);
-    setErrorMessage(null);
-    setSuccessMessage(null);
-    try {
-      await handleSubmit();
-    } finally {
-      setIsLoading(false);
-    }
-  }
-  {=/ isAnyPasswordBasedAuthEnabled =}
 
   // TODO(matija): this is called on every render, is it a problem?
   // If we do it in useEffect(), then there is a glitch between the default color and the
   // user provided one.
   const customTheme = createTheme(appearance ?? {})
 
-  const cta = isLogin ? 'Log in' : 'Sign up'
   const titles: Record<State, string> = {
```

Comment:
aha so `titles` is used both as titles but also as a state of the component? That was a bit hard to follow at first. maybe I'd make it more explicit (although I'm not sure how).

Reply:
> aha so titles is used both as titles but also as a state of the component? That was a bit hard to follow at first. maybe I'd make it more explicit (although I'm not sure how).

This says: "based on the State define a title" which means, for the "login" state, the title should be "Login" and so on. It's a common thing in Typescript to do it this way with `Record<SomeEnum, string>` which forces you to define the titles for each state (exhaustive pattern matching, similar to Haskell)